internal/app/pactproxy/proxy.go

```diff
@@ -25,157 +26,146 @@ var supportedMediaTypes = map[string]func([]byte, *url.URL) (requestDocument, er
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to find interaction for modifier. %s", modifier.Interaction))
 	}
 
 	log.Infof("adding modifier to interaction '%s'", interaction.Description)
 	interaction.Modifiers.AddModifier(modifier)
+
+	return c.NoContent(http.StatusOK)
 }
 
-func (a *api) sessionHandler(res http.ResponseWriter, req *http.Request) {
-	if req.Method == http.MethodDelete {
-		log.Infof("deleting session for %s", a.target)
-		a.proxy.ServeHTTP(res, req)
-		return
-	}
+func (a *api) sessionHandler(c echo.Context) error {
+	log.Infof("deleting session for %s", a.target)
+	return a.ProxyRequest(c)
 }
 
-func (a *api) interactionsHandler(res http.ResponseWriter, req *http.Request) {
-	if req.Method == http.MethodDelete {
-		log.Info("deleting interactions")
-		a.proxy.ServeHTTP(res, req)
-		a.interactions.Clear()
-		return
-	}
+func (a *api) interactionsDeleteHandler(c echo.Context) error {
+	log.Info("deleting interactions")
+	a.ProxyRequest(c)
+	a.interactions.Clear()
+	return nil
+}
 
-	if req.Method == http.MethodPost {
-		data, err := ioutil.ReadAll(req.Body)
-		if err != nil {
-			httpresponse.Errorf(res, http.StatusBadRequest, "unable to read interaction. %s", err.Error())
-			return
-		}
+func (a *api) interactionsPostHandler(c echo.Context) error {
+	data, err := ioutil.ReadAll(c.Request().Body)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to read interaction. %s", err.Error()))
+	}
 
-		interaction, err := LoadInteraction(data, req.URL.Query().Get("alias"))
-		if err != nil {
-			httpresponse.Errorf(res, http.StatusBadRequest, "unable to load interaction. %s", err.Error())
-			return
-		}
+	interaction, err := LoadInteraction(data, c.QueryParam("alias"))
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, httpresponse.Errorf("unable to load interaction. %s", err.Error()))
+	}
 
-		if interaction.Alias != "" {
-			log.Infof("storing interaction '%s' (%s)", interaction.Description, interaction.Alias)
-		} else {
-			log.Infof("storing interaction '%s'", interaction.Description)
-		}
+	if interaction.Alias != "" {
+		log.Infof("storing interaction '%s' (%s)", interaction.Description, interaction.Alias)
+	} else {
+		log.Infof("storing interaction '%s'", interaction.Description)
+	}
 
-		a.interactions.Store(interaction)
+	a.interactions.Store(interaction)
 
-		err = req.Body.Close()
-		if err != nil {
-			httpresponse.Errorf(res, http.StatusInternalServerError, err.Error())
-			return
-		}
+	err = c.Request().Body.Close()
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, httpresponse.Error(err.Error()))
+	}
 
-		req.Body = ioutil.NopCloser(bytes.NewBuffer(data))
+	c.Request().Body = ioutil.NopCloser(bytes.NewBuffer(data))
 
-		a.proxy.ServeHTTP(res, req)
-		return
-	}
+	return a.ProxyRequest(c)
 }
 
-func (a *api) interactionsWaitHandler(res http.ResponseWriter, req *http.Request) {
-	waitForCount, err := strconv.Atoi(req.URL.Query().Get("count"))
```

Comment:
Looks like we don't check the http method at all in this handler?  Easy to do, since it's not checked by the router